service/controller/clusterapi/controller.go

```diff
@@ -65,6 +66,12 @@ func NewController(config ControllerConfig) (*Controller, error) {
 		}
 	}
 
+	// We ensure the cluster-api controller is not checking reconciling the MC cluster CR to avoid conflicts.
```

Comment:
We could use a more elaborated comment here

```suggestion
	// Ensure cluster-api controller is not reconciling the MC cluster CR to avoid duplicate reconciliation with the management-cluster controller.
	// This due to CAPI installation providing a cluster CR for the MC.
```